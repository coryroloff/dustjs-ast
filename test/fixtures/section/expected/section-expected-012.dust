{+block}<p>{notbuffer}</p>{/block}