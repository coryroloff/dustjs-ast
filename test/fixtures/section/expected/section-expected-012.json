{
	"type": "Template",
	"body": [
		{
			"type": "Section",
			"body": [
				{
					"type": "Block",
					"body": [
						{
							"type": "Buffer",
							"value": "block",
							"start": 9,
							"end": 14,
							"loc": {
								"start": {
									"line": 1,
									"column": 9
								},
								"end": {
									"line": 1,
									"column": 14
								}
							}
						}
					],
					"start": 9,
					"end": 14,
					"loc": {
						"start": {
							"line": 1,
							"column": 9
						},
						"end": {
							"line": 1,
							"column": 14
						}
					}
				},
				{
					"type": "NamedBlock",
					"body": [
						{
							"type": "Buffer",
							"value": "else",
							"start": 21,
							"end": 25,
							"loc": {
								"start": {
									"line": 1,
									"column": 21
								},
								"end": {
									"line": 1,
									"column": 25
								}
							}
						}
					],
					"key": {
						"type": "Key",
						"name": "else",
						"start": 16,
						"end": 20,
						"loc": {
							"start": {
								"line": 1,
								"column": 16
							},
							"end": {
								"line": 1,
								"column": 20
							}
						}
					},
					"start": 14,
					"end": 25,
					"loc": {
						"start": {
							"line": 1,
							"column": 14
						},
						"end": {
							"line": 1,
							"column": 25
						}
					}
				},
				{
					"type": "NamedBlock",
					"body": [
						{
							"type": "Buffer",
							"value": "custom",
							"start": 34,
							"end": 40,
							"loc": {
								"start": {
									"line": 1,
									"column": 34
								},
								"end": {
									"line": 1,
									"column": 40
								}
							}
						}
					],
					"key": {
						"type": "Key",
						"name": "custom",
						"start": 27,
						"end": 33,
						"loc": {
							"start": {
								"line": 1,
								"column": 27
							},
							"end": {
								"line": 1,
								"column": 33
							}
						}
					},
					"start": 25,
					"end": 40,
					"loc": {
						"start": {
							"line": 1,
							"column": 25
						},
						"end": {
							"line": 1,
							"column": 40
						}
					}
				}
			],
			"params": [],
			"format": {
				"startTagAfterStart": null,
				"startTagBeforeParam": [],
				"startTagBeforeEnd": null,
				"endTagAfterStart": null,
				"endTagBeforeEnd": null
			},
			"kind": "?",
			"id": {
				"type": "Identifier",
				"path": [
					{
						"type": "Key",
						"name": "exists",
						"start": 2,
						"end": 8,
						"loc": {
							"start": {
								"line": 1,
								"column": 2
							},
							"end": {
								"line": 1,
								"column": 8
							}
						}
					}
				],
				"localContext": false,
				"start": 2,
				"end": 8,
				"loc": {
					"start": {
						"line": 1,
						"column": 2
					},
					"end": {
						"line": 1,
						"column": 8
					}
				}
			},
			"start": 0,
			"end": 49,
			"loc": {
				"start": {
					"line": 1,
					"column": 0
				},
				"end": {
					"line": 1,
					"column": 49
				}
			}
		}
	],
	"tokens": [
		{
			"type": "Punctuator",
			"value": "{",
			"start": 0,
			"end": 1,
			"loc": {
				"start": {
					"line": 1,
					"column": 0
				},
				"end": {
					"line": 1,
					"column": 1
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "?",
			"start": 1,
			"end": 2,
			"loc": {
				"start": {
					"line": 1,
					"column": 1
				},
				"end": {
					"line": 1,
					"column": 2
				}
			}
		},
		{
			"type": "Key",
			"value": "exists",
			"start": 2,
			"end": 8,
			"loc": {
				"start": {
					"line": 1,
					"column": 2
				},
				"end": {
					"line": 1,
					"column": 8
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 8,
			"end": 9,
			"loc": {
				"start": {
					"line": 1,
					"column": 8
				},
				"end": {
					"line": 1,
					"column": 9
				}
			}
		},
		{
			"type": "Buffer",
			"value": "block",
			"start": 9,
			"end": 14,
			"loc": {
				"start": {
					"line": 1,
					"column": 9
				},
				"end": {
					"line": 1,
					"column": 14
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "{",
			"start": 14,
			"end": 15,
			"loc": {
				"start": {
					"line": 1,
					"column": 14
				},
				"end": {
					"line": 1,
					"column": 15
				}
			}
		},
		{
			"type": "Punctuator",
			"value": ":",
			"start": 15,
			"end": 16,
			"loc": {
				"start": {
					"line": 1,
					"column": 15
				},
				"end": {
					"line": 1,
					"column": 16
				}
			}
		},
		{
			"type": "Key",
			"value": "else",
			"start": 16,
			"end": 20,
			"loc": {
				"start": {
					"line": 1,
					"column": 16
				},
				"end": {
					"line": 1,
					"column": 20
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 20,
			"end": 21,
			"loc": {
				"start": {
					"line": 1,
					"column": 20
				},
				"end": {
					"line": 1,
					"column": 21
				}
			}
		},
		{
			"type": "Buffer",
			"value": "else",
			"start": 21,
			"end": 25,
			"loc": {
				"start": {
					"line": 1,
					"column": 21
				},
				"end": {
					"line": 1,
					"column": 25
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "{",
			"start": 25,
			"end": 26,
			"loc": {
				"start": {
					"line": 1,
					"column": 25
				},
				"end": {
					"line": 1,
					"column": 26
				}
			}
		},
		{
			"type": "Punctuator",
			"value": ":",
			"start": 26,
			"end": 27,
			"loc": {
				"start": {
					"line": 1,
					"column": 26
				},
				"end": {
					"line": 1,
					"column": 27
				}
			}
		},
		{
			"type": "Key",
			"value": "custom",
			"start": 27,
			"end": 33,
			"loc": {
				"start": {
					"line": 1,
					"column": 27
				},
				"end": {
					"line": 1,
					"column": 33
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 33,
			"end": 34,
			"loc": {
				"start": {
					"line": 1,
					"column": 33
				},
				"end": {
					"line": 1,
					"column": 34
				}
			}
		},
		{
			"type": "Buffer",
			"value": "custom",
			"start": 34,
			"end": 40,
			"loc": {
				"start": {
					"line": 1,
					"column": 34
				},
				"end": {
					"line": 1,
					"column": 40
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "{",
			"start": 40,
			"end": 41,
			"loc": {
				"start": {
					"line": 1,
					"column": 40
				},
				"end": {
					"line": 1,
					"column": 41
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "/",
			"start": 41,
			"end": 42,
			"loc": {
				"start": {
					"line": 1,
					"column": 41
				},
				"end": {
					"line": 1,
					"column": 42
				}
			}
		},
		{
			"type": "Key",
			"value": "exists",
			"start": 42,
			"end": 48,
			"loc": {
				"start": {
					"line": 1,
					"column": 42
				},
				"end": {
					"line": 1,
					"column": 48
				}
			}
		},
		{
			"type": "Punctuator",
			"value": "}",
			"start": 48,
			"end": 49,
			"loc": {
				"start": {
					"line": 1,
					"column": 48
				},
				"end": {
					"line": 1,
					"column": 49
				}
			}
		}
	],
	"start": 0,
	"end": 49,
	"loc": {
		"start": {
			"line": 1,
			"column": 0
		},
		"end": {
			"line": 1,
			"column": 49
		}
	}
}